operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -645,134 +566,141 @@ func (r *RedpandaReconciler) clusterConfigFor(ctx context.Context, rp *redpandav
+	logger := log.FromContext(ctx).WithName(fmt.Sprintf("ClusterReconciler[%T].scaleDown", *cluster))
+	logger.V(TraceLevel).Info("starting StatefulSet scale down", "StatefulSet", client.ObjectKeyFromObject(set.StatefulSet).String())
+
+	brokerID, ok := brokerMap[set.LastPod.GetName()]
+	if ok {
+		// decommission if we have a brokerID, if not
+		// then the node has already been fully removed from
+		// the cluster and we can go ahead and delete the pod
+		// through patching the statefulset
 
-	log.V(logger.TraceLevel).Info(fmt.Sprintf("identified %d objects to gc", len(toDelete)))
+		requeue, err := r.decommissionBroker(ctx, admin, cluster, set, brokerID)
+		if err != nil {
+			return false, err
+		}
 
-	var errs []error
-	for _, obj := range toDelete {
-		if err := r.Client.Delete(ctx, obj); err != nil {
-			errs = append(errs, errors.Wrapf(err, "gc'ing %T: %s", obj, obj.GetName()))
+		if requeue {
+			return true, nil
 		}
 	}
 
-	return errors.Join(errs...)
-}
+	logger.V(TraceLevel).Info("scaling down StatefulSet", "StatefulSet", client.ObjectKeyFromObject(set.StatefulSet).String())
 
-func (r *RedpandaReconciler) needsDecommission(ctx context.Context, rp *redpandav1alpha2.Redpanda, stses []*appsv1.StatefulSet) (bool, error) {
-	client, err := r.ClientFactory.RedpandaAdminClient(ctx, rp)
-	if err != nil {
-		return false, errors.WithStack(err)
+	// now patch the statefulset to remove the pod
+	if err := r.LifecycleClient.PatchNodePoolSet(ctx, cluster, set.StatefulSet); err != nil {
+		return false, fmt.Errorf("scaling down statefulset: %w", err)
 	}
-	defer client.Close()
+	// we only do a statefulset at a time, waiting for them to
+	// become stable first
+	return true, nil
+}
+
+// decommissionBroker handles decommissioning a broker and waiting until it has finished decommissioning
+func (r *RedpandaReconciler) decommissionBroker(ctx context.Context, admin *rpadmin.AdminAPI, cluster *redpandav1alpha2.Redpanda, set *lifecycle.ScaleDownSet, brokerID int) (bool, error) {
+	logger := log.FromContext(ctx).WithName(fmt.Sprintf("ClusterReconciler[%T].decommissionBroker", *cluster))
+	logger.V(TraceLevel).Info("checking decommissioning status for pod", "Pod", client.ObjectKeyFromObject(set.LastPod).String())
 
-	health, err := client.GetHealthOverview(ctx)
+	decommissionStatus, err := admin.DecommissionBrokerStatus(ctx, brokerID)
 	if err != nil {
-		return false, errors.WithStack(err)
-	}
+		if strings.Contains(err.Error(), "is not decommissioning") {
+			logger.V(TraceLevel).Info("decommissioning broker", "Pod", client.ObjectKeyFromObject(set.LastPod).String())
+
+			if err := admin.DecommissionBroker(ctx, brokerID); err != nil {
+				return false, fmt.Errorf("decommissioning broker: %w", err)
+			}
 
-	desiredReplicas := 0
-	for _, sts := range stses {
-		desiredReplicas += int(ptr.Deref(sts.Spec.Replicas, 0))
+			return true, nil
+		} else {
+			return false, fmt.Errorf("fetching decommission status: %w", err)
```

Comment:
nit: use `errors.Wrapf` here as it includes stack traces which makes a HUGE difference when debugging.